### ISSUE TITLE
Strange problems persisting with HAX Reclaim install

### ISSUE THREAD

**Issue by lauragibbs** [elmsln/HAXcms#274]
I've got a new HAX install at Reclaim:
https://hax.lauragibbs.net/
Created a site called Stories
https://hax.lauragibbs.net/sites/stories/

I edited the default new page (renamed it Home, etc.), and I added a new page (S000001) and did extensive editing on that page. Now the dashboard says the site has two pages, but when I go to the site, there is only one page, and it has the old default name of the page (New page).

Since I cannot access the S000001 page, I'll hold off on doing any more work at the site for now; perhaps this is related to the symlink problem we were discussing at Twitter; here's that Twitter convo:
https://twitter.com/OnlineCrsLady/status/1138884314928689157 

Screenshots from this morning:
I
![Screen Shot 2019-06-13 at 10 10 10 AM](https://user-images.githubusercontent.com/19476543/59441369-79eb1000-8dc6-11e9-81e3-c4aee7d87af3.png)
![Screen Shot 2019-06-13 at 10 10 19 AM](https://user-images.githubusercontent.com/19476543/59441370-79eb1000-8dc6-11e9-97f2-7c12a77bb7df.png)

Available for any/all troubleshooting you want me to do on my end, but since Bryan and Tim mentioned some weirdness with the install, I'm guessing this might be something I cannot troubleshoot on my own. 

For what it's worth, things are weird when I look at the Stories site in Firefox; the browser error message I got yesterday (but Firefox version is v.67. Also, weirdly, the "Home" (instead of "New page") does show up there in Firefox. Screenshot:

![Screen Shot 2019-06-13 at 10 35 42 AM](https://user-images.githubusercontent.com/19476543/59441707-039add80-8dc7-11e9-8339-cb54b6a1afe9.png)

Thank you for your help!

**Comment by btopro** [comment]
very specific to reclaim hosting but I'll tag to keep note of. BTW thank for you the level of detail in this bug report it's perfect.

**Comment by btopro** [comment]
The site will work in Firefox 68 I bet but your right about there being a possible FF 67 issue which I've made a related issue for (published version of the site wouldn't have this issue but working on it locally would).

Helping already :p

**Comment by lauragibbs** [comment]
Super! When you get a chance to confer with Tim you can let me know if I should do another install or if I'll be able to pick up with the one at hax.lauragibbs.net ... it's fine with me either way. Starting tomorrow I'm creating a HAX stories column in the spreadsheet that rules my life this summer. Such perfect timing: I have exactly 40 days of summer left as of Friday, which means I can create a good heap of stories for HAXing.

**Comment by btopro** [comment]
tagging @timmmmyboy so this isn't just a weird twitter conversation

**Comment by timmmmyboy** [comment]
Thanks for the tag! To add to this I know there was a question of symlinks which worked on some of our servers but clearly didn't show up for Laura's install. I manually added those symlinks into @lauragibbs install when the conversation came up so if the issue identified in this bug report is persisting still then it is likely not related to the symlink problem.

**Comment by btopro** [comment]
it appears that the symlink has been removed again. Is there some kind of selinux or otherwise that would be killing this off?

Example:
Core code base has this css file -- https://hax.lauragibbs.net/build/es6/node_modules/@lrnwebcomponents/haxcms-elements/lib/base.css

Now in each site, they would be `/sites/stories/build` should be a symlink like `../../build build`
yet https://hax.lauragibbs.net/sites/stories/build/es6/node_modules/@lrnwebcomponents/haxcms-elements/lib/base.css
is not there

**Comment by timmmmyboy** [comment]
Interesting, it is indeed gone. I did put out 0.8.0 yesterday, but that shouldn't be touching that folder from my understanding. I'll do some digging to see what might be removing a symlink like that but for now @lauragibbs I have added the symlink back in.

**Comment by btopro** [comment]
@lauragibbs if you could make another site called test or something and then shoot a link here it might also be able to help @timmmmyboy and I figure out if this is some odd permissions internal thing or if haxcms is doing something silly.

**Comment by lauragibbs** [comment]
Got it! Iwent to hax.lauragibbs.net
I created site:
HAX Test
https://hax.lauragibbs.net/sites/hax-test/
I edited "New page" that shows up by default, but that change to the page name is not persisting, and even though the sites dashboard tells me I have multiple pages in the site, I cannot access them. Firefox continues to give me the "update browser" message, although I don't get that in Chrome. I hope that helps; these look like exactly the same things that were happening yesterday, so maybe that will make it easier to figure out and fix!
There are two "Page One" pages because I thought I had just forgotten to click save the first time, but instead I had hit save; I was just not seeing it come up in Chrome. I realized that when I created "Page One" again. So there are two Page Ones you can see there in the Firefox screenshot.

Screenshots:

FIREFOX (I'm not logged in)
![Screen Shot 2019-06-14 at 2 45 59 PM](https://user-images.githubusercontent.com/19476543/59531170-433ef380-8eb3-11e9-9a7a-e7a028ea649f.png)

CHROME (I'm logged in)
Dashboard shows multiple pages:
![Screen Shot 2019-06-14 at 2 45 24 PM](https://user-images.githubusercontent.com/19476543/59531191-518d0f80-8eb3-11e9-9d16-0a2305998d53.png)

Site view does not show those pages ( (and I had changed "New page" to "Testing Home" as you can see in the Firefox screenshot, but it shows up here still as "New page"):
![Screen Shot 2019-06-14 at 2 49 14 PM](https://user-images.githubusercontent.com/19476543/59531306-9ca72280-8eb3-11e9-9f1b-18ea55e4e656.png)

Even Outline does not show pages:
![Screen Shot 2019-06-14 at 2 47 52 PM](https://user-images.githubusercontent.com/19476543/59531267-826d4480-8eb3-11e9-92cf-293d51b88604.png)

**Comment by timmmmyboy** [comment]
Yep, same deal. No symlinks in that new site folder. At least we know it's reproduceable. I'll see if I can't trace what might be happening server side to not create those.

**Comment by btopro** [comment]
I setup a new account just now --
If you inspect the XHRs while doing things you'll see responses sometimes. here's the one I just got in trying to make a new site
```
<pre>Array
(
    [type] => 2
    [message] => symlink(): No such file or directory
    [file] => /home/therealc/public_html/haxcms/system/lib/HAXCMSSite.php
    [line] => 68
)
</pre>
```

**Comment by btopro** [comment]
`$_FILES['file-upload']` appears to also be empty, are there certain PHP settings that Cpanel blocks. Also what version of php does this end up coming with? I heard you mention PHP versions at the conference.

**Comment by btopro** [comment]
@lauragibbs if you make something and then it doesn't reflect the change on the next reload. Hold down `command + shift + R`. This triggers a "super refresh" that ignores local file copies. The data is still there it's something in the server sending headers to indicate this is a static file without changes (when its not).

**Comment by timmmmyboy** [comment]
I think Laura's .htaccess rules in her directory one level up we're trickling down and preventing the build folder from working (specifically Options -Indexes preventing the loading of the build directory). I moved the subdomain to use a folder outside of that and was able to create a test site and the symlinks are there. Also I see the symlinks worked automatically on the new account you created which is what I was seeing too. 

PHP is 7.0 by default with options to switch up to 7.3 (7.2 will be default at the end of July)

**Comment by btopro** [comment]
hmm mine might have been on 5.6 when I just logged in and set things up. I just changed it to PHP 7.3 and haxcms is way faster and uploads work (we don't support PHP 5.x currently so. duh on my part)
![2019-06-14_15-32-50](https://user-images.githubusercontent.com/329735/59533525-bc414980-8eb9-11e9-8bcc-134567b59b1f.png)

**Comment by btopro** [comment]
also php fpm is the bomb, that X needs to go away ;)

**Comment by btopro** [comment]
@lauragibbs if you change the PHP version in use to 7.3 it should help clear up some additional issues. I found something in HAXcms that prevents adding a site dashboard page from reloading automatically so fixing that currently

@timmmmyboy is there something in installatron or that I can do that would enforce it to install w/ at least php 7?

**Comment by btopro** [comment]
https://github.com/elmsln/HAXcms/releases/tag/0.8.1 fixes the sitelist dashboard issue

**Comment by lauragibbs** [comment]
Super! I'm not sure if I will have HAX time this weekend, but for sure on Monday. Thank you so much for ferreting out the problems. I'm not sure I know how to change the PHP version in use to 7.3. Is that something I do inside the Installatron panel? Or somewhere else in the Control Panel?
I'm a truly "tabula rasa" guinea pig who knows nothing and assumes nothing, ha ha. But I can follow orders! So if I need to manually change PHP version, just let me know how and I will do it. :-)

**Comment by btopro** [comment]
main page in reclaim apps area scroll down or search for this
![2019-06-14_15-57-27](https://user-images.githubusercontent.com/329735/59534761-3921f280-8ebd-11e9-91d9-027666c1dd9c.png)
then select the php version running and change it to this and hit apply
![2019-06-14_15-57-57](https://user-images.githubusercontent.com/329735/59534763-3a531f80-8ebd-11e9-97e4-e90e7b5cd79d.png)

**Comment by timmmmyboy** [comment]
0.8.2 will roll out to servers over the next 24 hours and the installer will start restricting to 7.0+ (it's unlikely your account would have been on 5.6, all of our servers default to 7, but figure it is a good idea to put that restriction in the installer).

**Comment by btopro** [comment]
Just checked on Symphony and I think I found the issue -- "PHP version: 7.1.3 or higher" so I bet I had it on 7 and that 7.1+ is required (which Drupal 8 and GravCMS would have as a requirement too I guess)

**Comment by btopro** [comment]
#263 added requirements to indicate 7.1 (didn't actually know that was the case, thought 7 was enough)

**Comment by lauragibbs** [comment]
WE'RE COOKING WITH GAS!!!!!! I changed to PHP 7.1 following @btopro 's instructions above, and things look GREAT! I created a new site, and it took just a few minutes to load up prepared texts (totally different experience than before; there were other weird things going on that I did not know were weird because I did not know what to expect), and then I used a spreadsheet to generate an index page. This is exactly what I needed: a totally text-focused way to reproduce public domain stories from full-text books online. Whoo-hoo!
https://hax.lauragibbs.net/sites/story-folder/
THANK YOU for all your help!!!!!!!!!! I'll write up a blog post later about the workflow, but this site now completes the last piece of the puzzle that I needed for this project. I should have 200 stories here by the end of the summer. And eventually........ thousands! **THANK YOU** to @btopro and @timmmmyboy 
Happy Monday!

**Comment by lauragibbs** [comment]
one more note for @btopro 
I tried downloading the site to view offline, and it gave me a nice zip folder which I unzipped, and I'm not able to view offline: should I write up a separate bug report for that? It's not urgent, but I wanted to let you know, esp. since it seems like it might be a related issue with this "update your browser" message. This is JUST THE DOWNLOADED SITE. So far, everything looks super for the actual published site in both Chrome and Firefox. 

![Screen Shot 2019-06-17 at 10 46 28 AM](https://user-images.githubusercontent.com/19476543/59613684-54734480-90ed-11e9-80d5-5efd79fff57f.png)
